### PR TITLE
Update to kubernetes-1.15.4

### PR DESCRIPTION
**Pull request by sslavic**
Towards giantswarm/giantswarm#6302

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -45,35 +45,35 @@ required = [
   name = "github.com/docker/distribution"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/apiextensions/pull/287 gets merged.

File: Gopkg.toml [hunk 2]
```diff
@@ -45,35 +45,35 @@ required = [
...
   name = "github.com/giantswarm/apiextensions"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/apprclient/pull/26 gets merged.

File: Gopkg.toml [hunk 3]
```diff
@@ -45,35 +45,35 @@ required = [
...
   name = "github.com/giantswarm/apprclient"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/e2e-harness/pull/209 gets merged.

File: Gopkg.toml [hunk 4]
```diff
@@ -45,35 +45,35 @@ required = [
...
   name = "github.com/giantswarm/e2e-harness"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/e2esetup/pull/37 gets merged.

File: Gopkg.toml [hunk 5]
```diff
@@ -45,35 +45,35 @@ required = [
...
   name = "github.com/giantswarm/e2esetup"
 
 [[constraint]]
   branch = "master"
   name = "github.com/giantswarm/e2etemplates"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/helmclient/pull/119 gets merged.

File: Gopkg.toml [hunk 6]
```diff
@@ -45,35 +45,35 @@ required = [
...
   name = "github.com/giantswarm/helmclient"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/k8sclient/pull/11 gets merged.

File: Gopkg.toml [hunk 7]
```diff
@@ -45,35 +45,35 @@ required = [
...
   name = "github.com/giantswarm/k8sclient"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/k8sportforward/pull/19 gets merged.

File: Gopkg.toml [hunk 8]
```diff
@@ -93,7 +93,7 @@ required = [
   name = "github.com/giantswarm/micrologger"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/operatorkit/pull/319 gets merged.

File: Gopkg.toml [hunk 9]
```diff
@@ -118,23 +118,43 @@ required = [
...
 [[constraint]]
   name = "k8s.io/apimachinery"
-  version = "kubernetes-1.13.2"
+  version = "kubernetes-1.15.4"
 
 [[constraint]]
+  name = "k8s.io/client-go"
+  version = "kubernetes-1.15.4"
+
+[[override]]
```
Comment (sslavic) [inline Gopkg.toml]: Without these overrides `dep ensure -update` falls into infinite loop or resolves wrong version (typically master).

Comment (rossf7) [reply Gopkg.toml]: Ack thanks for the explanation. That matches the problems I had when I worked on this.

**Pull request merged**